src/index.ts

```diff
@@ -10,31 +10,8 @@ import { PLUGIN_NAME, ABI_SUFFIX, DEFAULT_STARKNET_SOURCES_PATH, DEFAULT_STARKNE
 import { HardhatConfig, HardhatRuntimeEnvironment, HardhatUserConfig, HttpNetworkConfig } from "hardhat/types";
 import { adaptLog, adaptUrl, getDefaultHttpNetworkConfig } from "./utils";
 import { DockerWrapper, VenvWrapper } from "./starknet-wrappers";
+import { glob } from "hardhat/internal/util/glob";
 
-async function traverseFiles(
-    traversable: string,
-    predicate: (path: string) => boolean,
-    action: (path: string) => Promise<number>
-): Promise<number> {
-    let statusCode = 0;
-
-    const stats = fs.lstatSync(traversable);
-    if (stats.isDirectory()) {
-        for (const childName of fs.readdirSync(traversable)) {
-            const childPath = path.join(traversable, childName);
-            statusCode += await traverseFiles(childPath, predicate, action);
-        }
-    } else if (stats.isFile()) {
-        if (predicate(traversable)) {
-            statusCode += await action(traversable);
-        }
-    } else {
```

Comment:
As we discussed, these three cases should be handled (dir, file, other), the third case not being supported.

Reply:
To avoid duplicating this checking scheme, you could have a single function doing that (and the function could also accept the later used filter criterion).